compiler/noirc_errors/src/position.rs

```diff
@@ -6,7 +6,37 @@ use std::{
     ops::Range,
 };
 
-pub type Position = u32;
+#[derive(Copy, Clone, Default)]
+pub struct Position(u32, SrcId);
+
+impl Position {
+    pub fn new(pos: u32, src_id: SrcId) -> Position {
+        Position(pos, src_id)
+    }
+
+    pub fn src_id(&self) -> SrcId {
+        self.1
+    }
+}
+impl std::ops::Add<u32> for Position {
+    type Output = Position;
+
+    fn add(self, rhs: u32) -> Position {
+        Position(self.0 + rhs, self.1)
+    }
+}
+
+impl From<Position> for u32 {
+    fn from(val: Position) -> Self {
+        val.0
+    }
+}
```

Comment:
I think this impl is unclear in a struct with two Fields, even if one of those Fields is directly a `u32`. I'd prefer an explicit conversion like `position.index()` or position_index or similar.

Reply:
9e63896: chore: clearer Position naming